nautobot/extras/filters.py

```diff
@@ -1109,7 +1109,7 @@ class SecretFilterSet(
             "slug": "icontains",
         },
     )
-    # TODO dynamic choices needed
+    # 2.0 TODO(Glenn): dynamic choices needed. May be solved by `NaturalKeyorMultipleChoicePKFilter`?
```

Comment:
I don't see how NaturalKeyOrPKMultipleChoiceFilter would solve this - the issue is that secrets providers are Python classes, not database models.